flycheck.el

```diff
@@ -3447,8 +3447,14 @@ the THING at the column, and END the end of the THING."
             (eq mode 'lines))
         (flycheck--line-region beg)
       (or (pcase mode
-            (`symbols (flycheck-bounds-of-thing-at-point 'symbol beg))
-            (`sexps (flycheck-bounds-of-thing-at-point 'sexp beg)))
+            (`symbols
+             ;; Ensure that we're on a word or symbol.  See
+             ;; https://github.com/flycheck/flycheck/issues/1519
+             (and (< beg (point-max))
+                  (memq (char-syntax (char-after beg)) '(?w ?_))
+                  (flycheck-bounds-of-thing-at-point 'symbol beg)))
```

Comment:
What I had in mind (to avoid depending on the exact behavior of `bounds-of-thing-at-point`):
```suggestion
             (let ((bounds (flycheck-bounds-of-thing-at-point 'symbol beg)))
                 (when (and bounds (< beg (cdr bounds)))
                   bounds))
```
but this looks good too 👍

Reply:
Good point, I like your option too. It feels a tad cleaner to only call bounds-of-things-at-point when we're on a symbol constituent, but I don't have a strong opinion.